package/MDAnalysis/coordinates/DCD.py

```diff
@@ -522,26 +535,36 @@ def timeseries(self, asel, start=0, stop=-1, skip=1, format='afc'):
         # Check if the atom numbers can be grouped for efficiency, then we can read partial buffers
         # from trajectory file instead of an entire timestep
         # XXX needs to be implemented
-        return self._read_timeseries(atom_numbers, start, stop, skip, format)
+        return self._read_timeseries(atom_numbers, start, stop, step, format)
 
-    def correl(self, timeseries, start=0, stop=-1, skip=1):
+    def correl(self, timeseries, start=None, stop=None, step=None, skip=None):
```

Comment:
@orbeckst When you are saying 'behaves the way it should`, you mean that start stop and step reflect the same indices in a trajectory as numpy array slicing? If this is what you mean, It definitely does not, I can go in and change the pyrex code to do that in the same fashion that I did for `dcd.c`. With that being said, it is more complicated to test, I would have to think about going about doing that for a little bit.